web/base_viewer.js

```diff
@@ -1476,6 +1478,35 @@ class BaseViewer {
     this._setCurrentPageNumber(pageNumber, /* resetCurrentPageView = */ true);
     this.update();
   }
+
+  /**
+   * @type {number} One of the values in {ReadingDirection}.
+   */
+  get direction() {
+    return this._direction;
+  }
+
+  /**
+   * @param {number} dir
+   */
+  set direction(direction) {
+    if (this._direction === direction) {
+      return;
+    }
+    if (!Object.values(ReadingDirection).includes(direction)) {
```

Comment:
The validation needs to be done using the `isValidReadingDirection` helper function (see other review comment).